packages/core/src/shared/sam/utils.ts

```diff
@@ -19,6 +25,23 @@ import { RuntimeFamily, getFamily } from '../../lambda/models/samLambdaRuntime'
 export const getProjectRoot = (template: TemplateItem | undefined) =>
     template ? getProjectRootUri(template.uri) : undefined
 
+/**
+ * @description look for template in project root
+ * @param projectRoot  The URI of the root project folder
+ * @returns The URI of the template
+ * */
+export const getTemplateFile = async (projectRoot: vscode.Uri): Promise<vscode.Uri | undefined> => {
```

Comment:
We have `globals.templateRegistry` right?

Reply:
What is your suggestion?

Reply:
You are correct. We can use `getFile()` method like implement in `detectSamProject()`. However, that approach would still need no less modification than this. I see no extra value over the approach here.

Reply:
Developers copy existing code/patterns. If they see this, they will proliferate it. That's why it is valuable to stick to "one way to do things".

Reply:
I see. Giving that this PR should focus on fixing test, I'll be reverting the change here and will address this in separate PR.